packages/simple-xcm/simplexcm.ts

```diff
@@ -444,21 +462,31 @@ class PalletXcmBackend implements TransferBackend {
     const palletXcm = this.simpleXcm.api.tx[this.simpleXcm.palletXcm];
     const noXcmWeightLimit = 'Unlimited';
 
-    const txToDryRun = palletXcm.transferAssets(
-      destination,
-      beneficiary,
-      preparedParams.assets,
-      preparedParams.feeAssetIndex,
-      noXcmWeightLimit,
-    );
+    let estimatedFees;
+    do {
+      const txToDryRun = palletXcm.transferAssets(
+        destination,
+        beneficiary,
+        preparedParams.assets,
+        preparedParams.feeAssetIndex,
+        noXcmWeightLimit,
+      );
 
-    const estimatedFees = await this.simpleXcm.estimateExtrinsicXcmFees(
-      preparedParams.origin,
-      txToDryRun,
-      preparedParams.feeAssetId,
-    );
+      estimatedFees = await this.simpleXcm.tryEstimateExtrinsicXcmFees(
+        preparedParams.origin,
+        txToDryRun,
+        preparedParams.feeAssetId,
+      );
 
-    preparedParams.feeAnyAsset.fun.fungible += estimatedFees;
+      if ('error' in estimatedFees) {
+        if ('fungible' in preparedParams.feeAnyAssetRef.fun) {
```

Comment:
There is no need to check `fungible` existence here because `feeAnyAssetRef` has a `FungibleAnyAsset` type. It always has a `fungible` field

```suggestion
```